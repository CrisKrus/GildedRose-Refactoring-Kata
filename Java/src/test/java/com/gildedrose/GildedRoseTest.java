package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.assertEquals;

class GildedRoseTest {

    @Test
    void foo() {
        Item[] items = new Item[] {
            new Item("foo", 0, 0),
            new Item("foo", 5, 0),
            new Item("foo", 10, 0),
            new Item("foo", 0, 5),
            new Item("foo", 5, 5),
            new Item("foo", 10, 5),
        };
        GildedRose app = new GildedRose(items);
        app.updateQuality();

        Approvals.verifyAll("foo", Arrays.asList(items));
    }

}
