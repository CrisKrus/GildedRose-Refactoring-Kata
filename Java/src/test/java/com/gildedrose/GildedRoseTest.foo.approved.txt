foo[0] = foo, -1, 0
foo[1] = foo, 4, 0
foo[2] = foo, 9, 0
foo[3] = foo, -1, 3
foo[4] = foo, 4, 4
foo[5] = foo, 9, 4
